src/hotspot/share/opto/type.cpp

```diff
@@ -679,13 +680,17 @@ void Type::Initialize_shared(Compile* current) {
     TypeVect::VECTZ = TypeVect::make(T_FLOAT,16);
   }
 
+  if (Matcher::has_predicated_vectors()) {
+    TypeVMask::VMASK = TypeVMask::make(T_BYTE, Matcher::max_vector_size(T_BYTE));
+  }
+
```

Comment:
Instead of creating a new type altogether,  we can extended the mask type from the existing TypeVect class.
This is being done for X86 base patch, currently under review:
https://github.com/openjdk/jdk/pull/2768

I think same type (TypeVectMask) can be used for SVE support.

Reply:
Yes, there is some discussions focus on the mask type definition in jdk/jdk master. It's a good idea to define a subclass of `TypeVect`. It seems this can be used for SVE, but I'm not quite sure and we will have a try internally. Hope we can get a final point on the mask type first. Thanks!